client/keys/add.go

```diff
@@ -327,3 +336,32 @@ func printCreate(cmd *cobra.Command, info keyring.Info, showMnemonic bool, mnemo
 
 	return nil
 }
+
+func verifyMultisigTarget(kb keyring.Keyring, multisigKeys []string, newkey string) error {
+	kl, err := kb.List()
+	if err != nil {
+		return err
+	}
+
+	kmap := make(map[string]bool)
+	for i := 0; i < len(kl); i++ {
+		kmap[kl[i].GetName()] = true
+	}
+
+	if _, ok := kmap[newkey]; ok {
+		return errors.New("you cannot specify a new key as one of the names of the keys that make up a multisig")
+	}
```

Comment:
How about using `kr.Key(uid string)` or `kr.KeyByAddress(addr sdk.Address)` instead of this all this key traversal?